pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -388,24 +401,45 @@ private static Set<Integer> getSupportedBinaryOids() {
         Oid.UUID));
   }
 
-  private static Set<Integer> getBinaryOids(Properties info) throws PSQLException {
+  /**
+   * Gets all oids for which binary transfer can be enabled.
+   *
+   * @param info properties
+   * @return oids for which binary transfer can be enabled
+   * @throws PSQLException if any oid is not valid
+   */
+  private static Set<Integer> getBinaryEnabledOids(Properties info) throws PSQLException {
+    // check if binary transfer should be enabled for built-in types
     boolean binaryTransfer = PGProperty.BINARY_TRANSFER.getBoolean(info);
-    // Formats that currently have binary protocol support
+    // get formats that currently have binary protocol support
     Set<Integer> binaryOids = new HashSet<Integer>(32);
     if (binaryTransfer) {
       binaryOids.addAll(SUPPORTED_BINARY_OIDS);
     }
-
+    // add all oids which are enabled for binary transfer by the creator of the connection
     String oids = PGProperty.BINARY_TRANSFER_ENABLE.get(info);
     if (oids != null) {
       binaryOids.addAll(getOidSet(oids));
     }
-    oids = PGProperty.BINARY_TRANSFER_DISABLE.get(info);
+    return binaryOids;
+  }
+
+  /**
+   * Gets all oids for which binary transfer should be disabled.
+   *
+   * @param info properties
+   * @return oids for which binary transfer should be disabled or null, if no oids should be
+   *         disabled
+   * @throws PSQLException if any oid is not valid
+   */
+  @Nullable
+  private static Set<Integer> getBinaryDisabledOids(Properties info) throws PSQLException {
+    // check for oids that should explicitly be disabled
+    String oids = PGProperty.BINARY_TRANSFER_DISABLE.get(info);
     if (oids != null) {
-      binaryOids.removeAll(getOidSet(oids));
+      return new HashSet<Integer>(getOidSet(oids));
```

Comment:
Should this be something like `Collections.unmodifiableSet(...)`?

Reply:
I made it consistent with `getBinaryEnabledOids`, which also returns a modifiable set of values.
Actually this prevents the need to copy the set before modifying it (in the constructor of `PgConnection` the disabled OIDs are removed from the enabled OIDs to get the binary OIDs as a result).